app/components/SalesChart.tsx

```diff
@@ -0,0 +1,189 @@
+'use client';
+import React, { useState, useEffect } from 'react';
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend,
+} from 'chart.js';
+import { Bar } from 'react-chartjs-2';
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend
+);
+
+const chartOptions = {
+  responsive: true,
+  plugins: {
+    legend: {
+      position: 'top' as const,
+    },
+    title: {
+      display: true,
+      text: 'Ventas por Mes',
+    },
+  },
+};
+
+const months = [
+  'Enero',
+  'Febrero',
+  'Marzo',
+  'Abril',
+  'Mayo',
+  'Junio',
+  'Julio',
+  'Agosto',
+  'Septiembre',
+  'Octubre',
+  'Noviembre',
+  'Diciembre',
+];
+
+function GenerateRandomSales() {
+  return months.map(() => Math.floor(Math.random() * 10000) + 1000);
+}
+
+export function SalesChart() {
+  const currentYear = new Date().getFullYear();
+  const currentMonth = new Date().getMonth();
+
+  const [year, setYear] = useState(currentYear);
+  const [month, setMonth] = useState(currentMonth);
+  const [day, setDay] = useState(new Date().getDate());
+  const [minSales, setMinSales] = useState(0);
+  const [maxSales, setMaxSales] = useState(10000);
+  const [salesData, setSalesData] = useState({
+    labels: months,
+    datasets: [
+      {
+        label: `Ventas ${currentYear}`,
+        data: GenerateRandomSales(),
+        backgroundColor: 'rgba(53, 162, 235, 0.5)',
+      },
+    ],
+  });
+
+  useEffect(() => {
+    updateSales();
+  }, [year, month, day, minSales, maxSales]);
+
+  function updateSales() {
+    const newData = GenerateRandomSales().map((sale) =>
+      sale >= minSales && sale <= maxSales ? sale : 0
+    );
+
+    setSalesData({
+      labels: months,
+      datasets: [
+        {
+          label: `Ventas ${year}`,
+          data: newData,
+          backgroundColor: 'rgba(53, 162, 235, 0.5)',
+        },
+      ],
+    });
+  }
```

Comment:
this can be updated into a custom webhook and reuse it

Reply:
Resolved in commit 59b1a6e